### DISCUSSION TITLE
long poll for prefetching

### DISCUSSION THREAD

**Discussion by andrewnaeve** [TanStack/query#5762] [General]
For a normal query, to long poll I set the `refetchInterval`. However, this option is not available for prefetching. If I receive a response from my prefetch that indicates I want to check the server again, how do I do that?

**Comment by TkDodo** [comment]
I don't think that's what prefetching is about. If you get a response from prefetching that indicates that you should check the server again, that response is put into the cache for react-query. As soon as you call `useQuery` for that key (which will probably be quite soon after prefetching has happened), you'll have access to that data and can create a conditional `refetchInterval` with:

```
refetchInterval: (data) => // return a number based on data
```

this will work no matter how `data` got into the cache.

**Reply by andrewnaeve** [reply]
Thank you for the reply, I also did not know that `refetchInterval` took a callback. My use-case is that I have up to 20+ large reports in a tab browser, that are pre-fetched so that the user already has the report when they click on the tab. However, there is a "download all" button that assumes all reports are either fully fetched, or in process of being fetched. Would you opt for `useQueries` then for this?

**Reply by TkDodo** [reply]
probably
